From 0f2d9b683d2978e091dd79d81d19073dd8afaa62 Mon Sep 17 00:00:00 2001
From: "R, Kannappan" <r.kannappan@intel.com>
Date: Thu, 25 Jun 2015 19:50:26 +0530
Subject: [PATCH] fusb300: put the chip to low power mode during suspend

while entering suspend, put the chip to low power mode
by enabling power for wake monitoring circuit and while
resuming, enable power for measurement circuits.

Change-Id: I30a2eae015f619389af0c712ef60d28c305ef6a3
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-40217
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
Signed-off-by: Biju Chakkachattil <bijux.chakkachattil@intel.com>
---
 drivers/usb/typec/phy-fusb300.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/usb/typec/phy-fusb300.c b/drivers/usb/typec/phy-fusb300.c
index 33a4a7706bb2..1a1fe83e790e 100644
--- a/drivers/usb/typec/phy-fusb300.c
+++ b/drivers/usb/typec/phy-fusb300.c
@@ -1486,6 +1486,9 @@ static int fusb300_late_suspend(struct device *dev)
 
 	chip = dev_get_drvdata(dev);
 
+	/* enable power only for wakeup block */
+	regmap_write(chip->map, FUSB300_PWR_REG, FUSB300_PWR_BG_WKUP);
+
 	/* Disable the irq during suspend to prevent fusb300
 	isr executed before the i2c controller resume.*/
 	if (chip->client->irq) {
@@ -1502,6 +1505,11 @@ static int fusb300_early_resume(struct device *dev)
 
 	chip = dev_get_drvdata(dev);
 
+	/* enable the power for wakeup + measurement block */
+	regmap_write(chip->map, FUSB300_PWR_REG,
+			FUSB300_PWR_BG_WKUP | FUSB300_PWR_BMC |
+			FUSB300_PWR_MEAS);
+
 	/* Enable the irq after resume to prevent fusb300
 	isr executed before the i2c controller resume.*/
 	if (chip->client->irq) {
-- 
1.9.1

